packages/backend/src/execution/executors/QueryProviderExecutor.ts

```diff
@@ -1,22 +1,73 @@
 import { QueryProvider } from '../../QueryProvider';
 import { AnyQuery } from '../../types';
-import { QueryNode } from '../../query/createQueryTree';
 import { RefContext, createRefContext } from '../../refs/RefContext';
 import { QueryExecutor } from '../types';
 import { ColumnRef } from '../../refs/ColumnRef';
+import { Table, Where } from '@synthql/queries';
 
-export class QueryProviderExecutor implements QueryExecutor {
-    private providersByTable: Map<string, QueryProvider>;
-    constructor(providers: QueryProvider[]) {
+export class QueryProviderExecutor<DB> implements QueryExecutor {
+    private providersByTable: Map<string, QueryProvider<DB, Table<DB>>>;
+
+    constructor(providers: QueryProvider<DB, Table<DB>>[]) {
         this.providersByTable = new Map(providers.map((p) => [p.table, p]));
     }
 
+    extract(where: Where<DB, Table<DB>>) {
+        const w: Record<string, any> = {};
+
+        type N = any;
+
+        interface O {
+            in: Array<any>;
+        }
+
+        interface P {
+            '= any': any;
+        }
+
+        function isN(value: unknown): value is N {
```

Comment:
what is this cryptic language @jimezesinachi  😂  isN, isO?

Reply:
Just placeholder names. Just wanted to confirm that the implementation idea was solid before fleshing out. Will fix all of them now.